src/sentry/search/snuba/executors.py

```diff
@@ -1538,11 +1558,15 @@ def query(
                 Condition(Column("timestamp", joined_entity), Op.GTE, start),
                 Condition(Column("timestamp", joined_entity), Op.LT, end),
             ]
+            having = []
             for search_filter in search_filters or ():
                 # use the stored function if it exists in our mapping, otherwise use the basic lookup
                 fn = self.group_conditions_lookup.get(search_filter.key.name)
                 if fn:
                     where_conditions.append(fn(self, search_filter, joined_entity))
+                elif search_filter.key.name == "last_seen":
+                    # last seen modifies the having clause because its an aggregation
+                    having.append(self.get_last_seen_filter(search_filter, joined_entity))
```

Comment:
I think we could probably store metadata against the filters so that we know which clause they go into.

like
```
        "first_seen": SnubaFilter(get_first_seen_filter, FilterType.ROW),
        "last_seen": SnubaFilter(get_last_seen_filter, FilterType.AGGREGATE),
```
idk about the names but something like that

Reply:
Yea good call, let me think about how to do this